activiti-api-process-model/src/main/java/org/activiti/api/process/model/events/BPMNTimerExecutionFailureEvent.java

```diff
@@ -0,0 +1,21 @@
+/*
+ * Copyright 2018 Alfresco, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.activiti.api.process.model.events;
+
+public interface BPMNTimerExecutionFailureEvent extends BPMNTimerEvent {
```

Comment:
is it necessary to create an extra hierarchy level of interfaces with no methods as for BPMNTimerExecutionFailureEvent? Is there any code(maybe an another pr) where we check if it is bpm timer failure event?

As far as I remember it is definitely necessary to implement RuntimeEvent.

Reply:
@ffazzini
Implementation part of all these events will look like this:
https://github.com/Activiti/activiti-api/pull/121/commits/f44736f33bab8ffe840d2c7f992a487cb8d5735f#diff-b0e75f758e25bac91289de26355e9c8a

To get type of event (JOB_EXECUTION_FAILURE) we use this:
https://github.com/Activiti/activiti-api/pull/121/commits/f44736f33bab8ffe840d2c7f992a487cb8d5735f#diff-4b6ab325c9579f452f0e066a1b291a78R18

Reply:
ok so the design choice is consistent with current events design